src/components/Preferences/LocalIdentities.js

```diff
@@ -0,0 +1,250 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { format } from 'date-fns'
+import {
+  Badge,
+  Button,
+  IconCross,
+  IdentityBadge,
+  Info,
+  breakpoint,
+  font,
+  theme,
+} from '@aragon/ui'
+import { LocalIdentityModalContext } from '../LocalIdentityModal/LocalIdentityModalManager'
+import {
+  IdentityContext,
+  identityEventTypes,
+} from '../IdentityManager/IdentityManager'
+import EmptyLocalIdentities from './EmptyLocalIdentities'
+import Import from './Import'
+
+const LocalIdentities = ({
+  onClearAll,
+  onImport,
+  onModify,
+  onModifyEvent,
+  localIdentities,
+}) => {
+  // transform localIdentities from object into array
+  const identities = Object.keys(localIdentities).map(address => {
+    return Object.assign({}, localIdentities[address], { address })
+  })
+
+  if (!identities.length) {
+    return <EmptyLocalIdentities onImport={onImport} />
+  }
+  const { identityEvents$ } = React.useContext(IdentityContext)
+  const updateLabel = (fn, address) => async () => {
+    try {
+      await fn(address)
+      // preferences get all
+      onModifyEvent()
+      // for iframe apps
+      identityEvents$.next({ type: identityEventTypes.MODIFY, address })
+    } catch (e) {
+      /* nothing was updated */
+    }
+  }
+  const href = window.URL.createObjectURL(
+    new Blob([JSON.stringify(identities)], { type: 'text/json' })
+  )
+  // Mar 01 2019
+  const today = format(Date.now(), 'MMM dd yyyy')
+  const { showLocalIdentityModal } = React.useContext(LocalIdentityModalContext)
+
+  return (
+    <React.Fragment>
+      <Headers>
+        <div>Custom label</div>
+        <div>Address</div>
+      </Headers>
+      <List>
+        {identities.map(({ address, name }) => (
+          <Item key={address}>
+            <Label>{name}</Label>
+            <div>
+              <IdentityBadge
+                entity={address}
+                popoverAction={{
+                  label: 'Edit custom label',
+                  onClick: updateLabel(showLocalIdentityModal, address),
+                }}
+                popoverTitle={
+                  <PopoverActionTitle address={address} name={name} />
+                }
+              />
+            </div>
+          </Item>
+        ))}
+      </List>
+      <Controls>
+        <Import onImport={onImport} />
+        <StyledExport
+          label="Export labels"
+          mode="secondary"
+          download={`custom labels (${today}).json`}
```

Comment:
```suggestion
          download={`aragon-labels_${today}.json`}
```

IMO lets avoid creating files with spaces in the name, also I would make the format `mm-dd-yyyy` (03-26-2019)

Reply:
Agree on not using spaces. On dates, let's use [the standard](https://en.wikipedia.org/wiki/ISO_8601), which is `yyyy-mm-dd`